Sources/MapboxNavigation/Array.swift

```diff
@@ -32,3 +35,25 @@ extension Array where Element: NSAttributedString {
         return joinedAttributedString
     }
 }
+
+extension Array where Element == RouteStep {
+    // Find the longest contiguous series of RouteSteps connected to the first one.
+    //
+    // tolerance: -- Maximum distance between the end of one RouteStep and the start of the next to still consider them connected. Defaults to 100 meters
+    func continuousShape(tolerance: CLLocationDistance = 100) -> LineString? {
+        guard count > 0 else { return nil }
+        guard count > 1 else { return self[0].shape }
+
+        let filteredStepShapes = zip(compactMap { $0.shape }, suffix(from: 1).compactMap { $0.shape }).filter({
```

Comment:
Avoid mapping the array twice:

```suggestion
        let stepShapes = compactMap { $0.shape }
        let filteredStepShapes = zip(stepShapes, stepShapes.suffix(from: 1)).filter({
```